valora-inc/wallet
# feat(keyless-backup): exchange auth0 jwt for torus keyshare

### Description

Takes the credential you get from auth0 when you sign in with google and exchanges it for a torus keyshare. I had this take place as soon as login with google finishes, rather than after phone verification, since Torus rejects jwt's that are over a minute old, and it could easily take users a minute to verify their phone numbers. This will also reduce wait times after users complete phone verification since the Torus network is kinda slow.

This will let us encrypt and decrypt the mnemonic for keyless backup and recovery.

Not released to users yet

### Test plan

manually tested two flows:
- successful login: logged torus keyshare to console

https://github.com/valora-inc/wallet/assets/7979215/75483aa6-9fb0-4f59-9fca-0fdc29bf44e1

- simulated failure to login: made sure failure screen appears at the end

https://github.com/valora-inc/wallet/assets/7979215/72929460-47bb-4334-8c5e-8a96b8ac33e0

I also verified that Android and iOS give us the same keys for the same email (!!)

### Related issues

- Fixes https://linear.app/valora/issue/ACT-823/exchange-auth0-oauth-token-for-torus-private-key

### Backwards compatibility

na
